src/adapter/src/coord/sequencer.rs

```diff
@@ -979,64 +980,61 @@ impl<S: Append + 'static> Coordinator<S> {
             }
         };
 
+        // Knowing that we're only handling kafka sinks here helps us simplify.
+        let StorageSinkConnectionBuilder::Kafka(connection_builder) =
+            sink.connection_builder.clone();
+
         // Then try to create a placeholder catalog item with an unknown
         // connection. If that fails, we're done, though if the client specified
         // `if_not_exists` we'll tell the client we succeeded.
         //
         // This placeholder catalog item reserves the name while we create
         // the sink connection, which could take an arbitrarily long time.
-        let op = catalog::Op::CreateItem {
+
+        let catalog_sink = catalog::Sink {
+            create_sql: sink.create_sql,
+            from: sink.from,
+            connection: StorageSinkConnectionState::Pending(StorageSinkConnectionBuilder::Kafka(
+                connection_builder,
+            )),
+            envelope: sink.envelope,
+            with_snapshot,
+            depends_on,
+            compute_instance,
```

Comment:
We shouldn't really need a `compute_instance` on sinks anymore. Is that only because of how the `index_oracle` thing works?